data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/configuration/IndexParametersConfiguration.java

```diff
@@ -5,30 +5,50 @@
 
 package org.opensearch.dataprepper.plugins.source.opensearch.configuration;
 
+import com.fasterxml.jackson.annotation.JsonIgnore;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.AssertTrue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collections;
 import java.util.List;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+import java.util.stream.Collectors;
 
 public class IndexParametersConfiguration {
 
+    private static final Logger LOG = LoggerFactory.getLogger(IndexParametersConfiguration.class);
+
     @JsonProperty("include")
-    private List<String> include;
+    @Valid
+    private List<IncludedIndex> include = Collections.emptyList();
 
     @JsonProperty("exclude")
-    private List<String> exclude;
+    private List<String> exclude = Collections.emptyList();
 
-    public void setInclude(List<String> include) {
-        this.include = include;
-    }
+    @JsonIgnore
+    private List<Pattern> excludePatterns = Collections.emptyList();
 
-    public void setExclude(List<String> exclude) {
-        this.exclude = exclude;
+    public List<IncludedIndex> getIncludedIndices() {
+        return include;
     }
 
-    public List<String> getInclude() {
-        return include;
+    public List<Pattern> getExcludePatterns() {
+        return excludePatterns;
     }
 
-    public List<String> getExclude() {
-        return exclude;
+    @AssertTrue(message = "one of the exclude regex patterns is invalid")
+    boolean isExcludeRegexValid() {
+        try {
+            excludePatterns = exclude.stream().map(Pattern::compile).collect(Collectors.toList());
+            return true;
+        } catch (final PatternSyntaxException e){
+            LOG.error("Invalid exclude regex pattern: ", e);
```

Comment:
Rather than include the stack trace (since this is a validation error, not an unexpected exception), I'd say put these into a loop where you catch each individual pattern. Then you can log the pattern.

```
LOG.error("Invalid exclude regex pattern: '{}'", pattern);
```

Reply:
Will put in a loop